xds/src/main/java/io/grpc/xds/internal/sds/ServerWrapperForXds.java

```diff
@@ -36,22 +44,62 @@
 public final class ServerWrapperForXds extends Server {
   private final Server delegate;
   private final XdsClientWrapperForServerSds xdsClientWrapperForServerSds;
-
-  ServerWrapperForXds(Server delegate, XdsClientWrapperForServerSds xdsClientWrapperForServerSds) {
+  @Nullable XdsServerBuilder.ErrorNotifier errorNotifier;
+  @Nullable XdsClientWrapperForServerSds.ServerWatcher serverWatcher;
+  @GuardedBy("this") private boolean started;
+
+  ServerWrapperForXds(
+      Server delegate,
+      XdsClientWrapperForServerSds xdsClientWrapperForServerSds,
```

Comment:
This doesn't look right. This means the xds client is running and doing I/O before start(). It should be created within start(). Probably best not to fix it in this PR though.

Reply:
This is behaving as expected. The xdsClient is only started during `ServerWrapperForXds.start()` as you can see the call to `xdsClientWrapperForServerSds.createXdsClientAndStart()` inside `ServerWrapperForXds.start()`. Only for unit tests a mocked XdsClient is injected into `XdsClientWrapperForServerSds` that doesn't need to be started and that's the reason for the `if (!xdsClientWrapperForServerSds.hasXdsClient())` clause.

Reply:
Hmm... that's problematic as if start() throws we need to call shutdown() on it. But can it be restarted?

Reply:
Hmmm. So what's the general behavior of a `Server` in gRPC? If `start()` throws, is it possible for the user to retry `start` (depending on the nature of exception) or do we ask them to just build a new server and try start on that? Depending on that I can fix this start so the xDS client can be shutdown in case of an error and make `start()` retriable. I don't see any reason why retry cannot be made to work but depends on what we want.

Reply:
>If start() throws, is it possible for the user to retry start (depending on the nature of exception) or do we ask them to just build a new server and try start on that?

This is related to the topic proposed in out ExperimentalApi meeting about if `start()` throws `IOException` should the library automatically clean up all resources by its spec.

Reply:
For now this one doesn't do any cleanup when `start` throws. Can revisit this in a separate PR (along with the other `start` related issue mentioned?)